File: backend/supabase_client.py
import os
from dotenv import load_dotenv
from supabase import create_client, Client

# Load environment variables from .env file
load_dotenv()

SUPABASE_URL = os.environ.get("SUPABASE_URL")
SUPABASE_KEY = os.environ.get("SUPABASE_KEY")

def get_supabase_client() -> Client:
    """
    Initialize and return the Supabase client.
    Ensure that SUPABASE_URL and SUPABASE_KEY are set in your environment or .env file.
    """
    if not SUPABASE_URL or not SUPABASE_KEY:
        raise ValueError("Supabase credentials not found. Please set SUPABASE_URL and SUPABASE_KEY.")
    
    return create_client(SUPABASE_URL, SUPABASE_KEY)

File: supabase-client.js
/* supabase-client.js */

const SUPABASE_URL = "sb_publishable_BkNTJMFpBmxuLZMr8BGYIA_kUpC_qcR";
const SUPABASE_KEY = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6InN5eHVndXhobmFvcWFmcmVrbGJyIiwicm9sZSI6InNlcnZpY2Vfcm9sZSIsImlhdCI6MTc4MTc5NTg1NSwiZXhwIjoyMDk3MzcxODU1fQ.CRYfs4nyIirPkM11ZPiid76iqNzUCrCBLOOGbFg0e6A";

window.supabaseClient = window.supabase.createClient(
    SUPABASE_URL,
    SUPABASE_KEY
);